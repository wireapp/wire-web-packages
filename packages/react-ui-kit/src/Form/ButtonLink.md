Demo:

```js
import {ButtonLink, Container, Columns, Column, COLOR} from '@wireapp/react-ui-kit';

<Container>
  <Columns>
    <Column>ButtonLink</Column>
    <Column>
      <ButtonLink backgroundColor={COLOR.GREEN}>ButtonLink</ButtonLink>
    </Column>
  </Columns>
  <Columns>
    <Column>Loading ButtonLink</Column>
    <Column>
      <ButtonLink showLoading backgroundColor={COLOR.GREEN}>
        ButtonLink
      </ButtonLink>
    </Column>
  </Columns>
</Container>;
```
