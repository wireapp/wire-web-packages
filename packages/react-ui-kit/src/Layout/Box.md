Demo:

```js
import {Box} from '@wireapp/react-ui-kit';

<Box>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce a mattis nibh, sed maximus leo. Fusce a lacinia sem,
  vitae ornare dolor. Quisque rhoncus, magna non lacinia sagittis, erat augue fringilla metus, eu consectetur leo velit
  non lacus. Phasellus ipsum turpis, dapibus ut purus in, lobortis consectetur mi.
</Box>;
```
