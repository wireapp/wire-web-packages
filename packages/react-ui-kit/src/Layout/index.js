import setGlobalStyles from '../globalStyles';

setGlobalStyles();

export * from './Column';
export * from './Container';
export * from './Content';
export * from './Footer';
export * from './Header';
export * from './StyledApp';
