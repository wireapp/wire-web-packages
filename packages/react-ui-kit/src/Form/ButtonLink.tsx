/*
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

/** @jsx jsx */
import {ObjectInterpolation, jsx} from '@emotion/core';
import {Loading} from '../Misc';
import {COLOR} from '../Identity';
import {filterProps} from '../util';
import {filterTextProps} from '../Text';
import {ButtonProps, buttonStyle} from './Button';
import {Theme} from '../Layout';

export const buttonLinkStyle: (
  theme: Theme,
  props: ButtonProps<HTMLAnchorElement>,
) => ObjectInterpolation<undefined> = (theme, props) => ({
  ...buttonStyle(theme, props),
  display: 'inline-flex !important',
});

const filterButtonLinkProps = (props: ButtonProps<HTMLAnchorElement>) => {
  return filterProps(filterTextProps(props) as ButtonProps<HTMLAnchorElement>, [
    'backgroundColor',
    'disabled',
    'noCapital',
  ]);
};

export const ButtonLink = ({
  children,
  showLoading,
  loadingColor = COLOR.WHITE,
  ...props
}: ButtonProps<HTMLAnchorElement>) => (
  <a css={theme => buttonLinkStyle(theme, props)} {...filterButtonLinkProps(props)}>
    {showLoading ? <Loading size={30} color={loadingColor} style={{display: 'flex', margin: 'auto'}} /> : children}
  </a>
);
