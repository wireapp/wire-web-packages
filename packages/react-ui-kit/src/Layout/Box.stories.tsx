/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {Meta, StoryObj} from '@storybook/react';

import {Box} from './Box';

import {COLOR} from '../Identity/colors';
import {Text} from '../Text';

const meta: Meta<typeof Box> = {
  component: Box,
  title: 'Layout/Box',
  decorators: [
    Story => (
      <div style={{padding: '24px', maxWidth: '600px', margin: '0 auto'}}>
        <Story />
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Box>;

const LOREM_IPSUM = `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce a mattis nibh, sed maximus leo. Fusce a lacinia sem,
vitae ornare dolor. Quisque rhoncus, magna non lacinia sagittis, erat augue fringilla metus, eu consectetur leo velit
non lacus. Phasellus ipsum turpis, dapibus ut purus in, lobortis consectetur mi.`;

export const Default: Story = {
  args: {
    children: LOREM_IPSUM,
  },
};

export const WithCustomPadding: Story = {
  args: {
    children: LOREM_IPSUM,
    style: {padding: '32px'},
  },
};

export const WithCustomBorder: Story = {
  args: {
    children: LOREM_IPSUM,
    style: {border: `2px solid ${COLOR.BLUE}`},
  },
};

export const WithFormattedContent: Story = {
  args: {
    children: (
      <>
        <Text block style={{fontSize: '20px', fontWeight: 'bold', marginBottom: '16px'}}>
          Important Notice
        </Text>
        <Text block style={{marginBottom: '12px'}}>
          {LOREM_IPSUM}
        </Text>
        <Text block style={{color: COLOR.GRAY}}>
          Last updated: 2025-01-01
        </Text>
      </>
    ),
  },
};

export const Nested: Story = {
  args: {
    children: (
      <>
        <Text block style={{marginBottom: '16px'}}>
          Outer box content
        </Text>
        <Box style={{backgroundColor: COLOR.GRAY_LIGHTEN_92}}>
          <Text>Inner box content</Text>
        </Box>
      </>
    ),
  },
};

export const Interactive: Story = {
  args: {
    children: (
      <div style={{textAlign: 'center'}}>
        <Text block style={{marginBottom: '16px'}}>
          Click the button below
        </Text>
        <button
          style={{
            backgroundColor: COLOR.BLUE,
            border: 'none',
            borderRadius: '4px',
            color: COLOR.WHITE,
            cursor: 'pointer',
            padding: '8px 16px',
          }}
        >
          Action Button
        </button>
      </div>
    ),
    style: {cursor: 'pointer'},
  },
};
