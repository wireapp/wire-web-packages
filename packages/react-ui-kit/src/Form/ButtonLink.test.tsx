/*
 * Wire
 * Copyright (C) 2019 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import React from 'react';
import {THEME_ID} from '../Layout';
import {matchComponent} from '../test/testUtil';
import {ButtonLink} from './ButtonLink';

describe('"ButtonLink"', () => {
  it('renders', () => matchComponent(<ButtonLink>Submit</ButtonLink>));
  it('renders (dark theme)', () => matchComponent(<ButtonLink>ButtonLink</ButtonLink>, THEME_ID.DARK));
  it('renders in loading state', () => matchComponent(<ButtonLink showLoading>Submit</ButtonLink>));
});
