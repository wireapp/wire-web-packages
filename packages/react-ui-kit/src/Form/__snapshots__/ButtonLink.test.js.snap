// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"ButtonLink" renders (dark theme) 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1mc7zwm-StyledAppContainer{background:#000;-webkit-transition:background 0.15s;transition:background 0.15s;}",
      }
    }
    data-emotion="css 1mc7zwm-StyledAppContainer"
  />,
  <div
    className="css-1mc7zwm-StyledAppContainer"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": "html{line-height:1.15;-webkit-text-size-adjust:100%;}body{margin:0;}main{display:block;}h1{font-size:2em;margin:0.67em 0;}hr{box-sizing:content-box;height:0;overflow:visible;}pre{font-family:monospace,monospace;font-size:1em;}a{background-color:transparent;}abbr[title]{border-bottom:none;-webkit-text-decoration:underline;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted;}b,strong{font-weight:bolder;}code,kbd,samp{font-family:monospace,monospace;font-size:1em;}small{font-size:80%;}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline;}sub{bottom:-0.25em;}sup{top:-0.5em;}img{border-style:none;}button,input,optgroup,select,textarea{font-family:inherit;font-size:100%;line-height:1.15;margin:0;}button,input{overflow:visible;}button,select{text-transform:none;}button,[type="button"],[type="reset"],[type="submit"]{-webkit-appearance:button;}button::-moz-focus-inner,[type="button"]::-moz-focus-inner,[type="reset"]::-moz-focus-inner,[type="submit"]::-moz-focus-inner{border-style:none;padding:0;}button:-moz-focusring,[type="button"]:-moz-focusring,[type="reset"]:-moz-focusring,[type="submit"]:-moz-focusring{outline:1px dotted ButtonText;}fieldset{padding:0.35em 0.75em 0.625em;}legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal;}progress{vertical-align:baseline;}textarea{overflow:auto;}[type="checkbox"],[type="radio"]{box-sizing:border-box;padding:0;}[type="number"]::-webkit-inner-spin-button,[type="number"]::-webkit-outer-spin-button{height:auto;}[type="search"]{-webkit-appearance:textfield;outline-offset:-2px;}[type="search"]::-webkit-search-decoration{-webkit-appearance:none;}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit;}details{display:block;}summary{display:-webkit-box;display:-webkit-list-item;display:-ms-list-itembox;display:list-item;}template{display:none;}[hidden]{display:none;}*{box-sizing:border-box;}b,strong{font-weight:600;}body{--accent-color:#0667c8;-webkit---accent-color-highlight:#e7f0fa;--accent-color-highlight:#e7f0fa;--accent-color-highlight-inversed:#022950;--accent-color-border:#0667c8;--accent-color-focus:#3885d3;--icon-primary-active-fill:#0667c8;--icon-secondary-active-border:transparent;--button-primary-hover:#0552a0;--button-primary-active:#043e78;--button-primary-active-border:#043e78;--button-primary-focus-border:#043e78;-moz-osx-font-smoothing:grayscale;-webkit-font-smoothing:antialiased;background:#000;color:#fff;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Helvetica,Arial,sans-serif;font-weight:300;line-height:1.5;min-height:100vh;-webkit-transition:background 0.15s;transition:background 0.15s;}body,body.theme-default{--checkbox-background:#edeff0;--checkbox-background-disabled:#fafafa;--checkbox-background-disabled-selected:#9fa1a7;--checkbox-border:#54585f;--checkbox-border-disabled:#676b71;--icon-button-primary-enabled-bg:#fff;--icon-button-primary-hover-bg:#edeff0;--icon-button-primary-border:#dce0e3;--icon-button-primary-disabled-bg:#edeff0;--icon-button-primary-disabled-border:#dce0e3;--icon-button-primary-hover-border:#cbced1;--button-primary-disabled-bg:#cbced1;--button-primary-disabled-text:#54585f;--button-tertiary-bg:#fff;--button-tertiary-border:#dce0e3;--button-tertiary-hover-bg:#edeff0;--button-tertiary-hover-border:#cbced1;--button-tertiary-disabled-bg:#edeff0;--button-tertiary-disabled-border:transparent;--text-input-background:#fff;--text-input-border:#dce0e3;--text-input-border-hover:#9fa1a7;--text-input-placeholder:#676b71;--text-input-disabled:#edeff0;--text-input-label:#54585f;--select-focused-description:#fff;--danger-color:#c20013;--app-bg:#fafafa;--main-color:#000;}body.theme-dark{--checkbox-background:#34373d;--checkbox-background-disabled:#34373d;--checkbox-background-disabled-selected:#54585f;--checkbox-border:#9fa1a7;--checkbox-border-disabled:#9fa1a7;--icon-button-primary-enabled-bg:#34373d;--icon-button-primary-hover-bg:#54585f;--icon-button-primary-border:#17181a;--icon-button-primary-disabled-bg:#26272c;--icon-button-primary-disabled-border:#34373d;--icon-button-primary-hover-border:#676b71;--button-primary-disabled-bg:#676b71;--button-primary-disabled-text:#000;--button-tertiary-bg:#34373d;--button-tertiary-border:#17181a;--button-tertiary-hover-bg:#54585f;--button-tertiary-hover-border:#54585f;--button-tertiary-disabled-bg:#26272c;--button-tertiary-disabled-border:transparent;--text-input-background:#000;--text-input-border:#54585f;--text-input-border-hover:#dce0e3;--text-input-placeholder:#9fa1a7;--text-input-disabled:#17181a;--text-input-label:#dce0e3;--select-focused-description:#dce0e3;--danger-color:#ff7770;--app-bg:#26272c;--main-color:#fff;}html{background:#000;-webkit-transition:background 0.15s;transition:background 0.15s;}p{margin-top:0;}",
        }
      }
      data-emotion="css-global 1vzghze-#54585f-#dce0e3-globalStyle"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-kjqv3l-ButtonLink{color:var(--text-input-background);display:-webkit-inline-box!important;display:-webkit-inline-flex!important;display:-ms-inline-flexbox!important;display:inline-flex!important;font-size:16px;font-weight:600;overflow:hidden;text-align:center;text-overflow:ellipsis;text-transform:none;white-space:nowrap;border:0;cursor:pointer;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;margin-bottom:16px;padding:0 16px;outline:none;-webkit-text-decoration:none;text-decoration:none;touch-action:manipulation;-webkit-transition:all 0.24s;transition:all 0.24s;width:auto;border-radius:16px;height:48px;line-height:48px;max-width:100%;min-width:125px;background-color:var(--accent-color);}.css-kjqv3l-ButtonLink:hover,.css-kjqv3l-ButtonLink:focus{background-color:var(--button-primary-hover);}.css-kjqv3l-ButtonLink:focus{border:1px solid var(--button-primary-focus-border);}.css-kjqv3l-ButtonLink:active{background-color:var(--button-primary-active);border:1px solid var(--button-primary-active-border);color:#fff;}",
        }
      }
      data-emotion="css kjqv3l-ButtonLink"
    />
    <a
      className="css-kjqv3l-ButtonLink"
    >
      ButtonLink
    </a>
  </div>,
]
`;

exports[`"ButtonLink" renders 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-7uml9n-StyledAppContainer{background:#edeff0;-webkit-transition:background 0.15s;transition:background 0.15s;}",
      }
    }
    data-emotion="css 7uml9n-StyledAppContainer"
  />,
  <div
    className="css-7uml9n-StyledAppContainer"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": "html{line-height:1.15;-webkit-text-size-adjust:100%;}body{margin:0;}main{display:block;}h1{font-size:2em;margin:0.67em 0;}hr{box-sizing:content-box;height:0;overflow:visible;}pre{font-family:monospace,monospace;font-size:1em;}a{background-color:transparent;}abbr[title]{border-bottom:none;-webkit-text-decoration:underline;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted;}b,strong{font-weight:bolder;}code,kbd,samp{font-family:monospace,monospace;font-size:1em;}small{font-size:80%;}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline;}sub{bottom:-0.25em;}sup{top:-0.5em;}img{border-style:none;}button,input,optgroup,select,textarea{font-family:inherit;font-size:100%;line-height:1.15;margin:0;}button,input{overflow:visible;}button,select{text-transform:none;}button,[type="button"],[type="reset"],[type="submit"]{-webkit-appearance:button;}button::-moz-focus-inner,[type="button"]::-moz-focus-inner,[type="reset"]::-moz-focus-inner,[type="submit"]::-moz-focus-inner{border-style:none;padding:0;}button:-moz-focusring,[type="button"]:-moz-focusring,[type="reset"]:-moz-focusring,[type="submit"]:-moz-focusring{outline:1px dotted ButtonText;}fieldset{padding:0.35em 0.75em 0.625em;}legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal;}progress{vertical-align:baseline;}textarea{overflow:auto;}[type="checkbox"],[type="radio"]{box-sizing:border-box;padding:0;}[type="number"]::-webkit-inner-spin-button,[type="number"]::-webkit-outer-spin-button{height:auto;}[type="search"]{-webkit-appearance:textfield;outline-offset:-2px;}[type="search"]::-webkit-search-decoration{-webkit-appearance:none;}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit;}details{display:block;}summary{display:-webkit-box;display:-webkit-list-item;display:-ms-list-itembox;display:list-item;}template{display:none;}[hidden]{display:none;}*{box-sizing:border-box;}b,strong{font-weight:600;}body{--accent-color:#0667c8;-webkit---accent-color-highlight:#e7f0fa;--accent-color-highlight:#e7f0fa;--accent-color-highlight-inversed:#022950;--accent-color-border:#0667c8;--accent-color-focus:#3885d3;--icon-primary-active-fill:#0667c8;--icon-secondary-active-border:transparent;--button-primary-hover:#0552a0;--button-primary-active:#043e78;--button-primary-active-border:#043e78;--button-primary-focus-border:#043e78;-moz-osx-font-smoothing:grayscale;-webkit-font-smoothing:antialiased;background:#edeff0;color:rgb(29, 30, 31);display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Helvetica,Arial,sans-serif;font-weight:300;line-height:1.5;min-height:100vh;-webkit-transition:background 0.15s;transition:background 0.15s;}body,body.theme-default{--checkbox-background:#edeff0;--checkbox-background-disabled:#fafafa;--checkbox-background-disabled-selected:#9fa1a7;--checkbox-border:#54585f;--checkbox-border-disabled:#676b71;--icon-button-primary-enabled-bg:#fff;--icon-button-primary-hover-bg:#edeff0;--icon-button-primary-border:#dce0e3;--icon-button-primary-disabled-bg:#edeff0;--icon-button-primary-disabled-border:#dce0e3;--icon-button-primary-hover-border:#cbced1;--button-primary-disabled-bg:#cbced1;--button-primary-disabled-text:#54585f;--button-tertiary-bg:#fff;--button-tertiary-border:#dce0e3;--button-tertiary-hover-bg:#edeff0;--button-tertiary-hover-border:#cbced1;--button-tertiary-disabled-bg:#edeff0;--button-tertiary-disabled-border:transparent;--text-input-background:#fff;--text-input-border:#dce0e3;--text-input-border-hover:#9fa1a7;--text-input-placeholder:#676b71;--text-input-disabled:#edeff0;--text-input-label:#54585f;--select-focused-description:#fff;--danger-color:#c20013;--app-bg:#fafafa;--main-color:#000;}body.theme-dark{--checkbox-background:#34373d;--checkbox-background-disabled:#34373d;--checkbox-background-disabled-selected:#54585f;--checkbox-border:#9fa1a7;--checkbox-border-disabled:#9fa1a7;--icon-button-primary-enabled-bg:#34373d;--icon-button-primary-hover-bg:#54585f;--icon-button-primary-border:#17181a;--icon-button-primary-disabled-bg:#26272c;--icon-button-primary-disabled-border:#34373d;--icon-button-primary-hover-border:#676b71;--button-primary-disabled-bg:#676b71;--button-primary-disabled-text:#000;--button-tertiary-bg:#34373d;--button-tertiary-border:#17181a;--button-tertiary-hover-bg:#54585f;--button-tertiary-hover-border:#54585f;--button-tertiary-disabled-bg:#26272c;--button-tertiary-disabled-border:transparent;--text-input-background:#000;--text-input-border:#54585f;--text-input-border-hover:#dce0e3;--text-input-placeholder:#9fa1a7;--text-input-disabled:#17181a;--text-input-label:#dce0e3;--select-focused-description:#dce0e3;--danger-color:#ff7770;--app-bg:#26272c;--main-color:#fff;}html{background:#edeff0;-webkit-transition:background 0.15s;transition:background 0.15s;}p{margin-top:0;}",
        }
      }
      data-emotion="css-global 11ozpjr-#54585f-#dce0e3-globalStyle"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-kjqv3l-ButtonLink{color:var(--text-input-background);display:-webkit-inline-box!important;display:-webkit-inline-flex!important;display:-ms-inline-flexbox!important;display:inline-flex!important;font-size:16px;font-weight:600;overflow:hidden;text-align:center;text-overflow:ellipsis;text-transform:none;white-space:nowrap;border:0;cursor:pointer;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;margin-bottom:16px;padding:0 16px;outline:none;-webkit-text-decoration:none;text-decoration:none;touch-action:manipulation;-webkit-transition:all 0.24s;transition:all 0.24s;width:auto;border-radius:16px;height:48px;line-height:48px;max-width:100%;min-width:125px;background-color:var(--accent-color);}.css-kjqv3l-ButtonLink:hover,.css-kjqv3l-ButtonLink:focus{background-color:var(--button-primary-hover);}.css-kjqv3l-ButtonLink:focus{border:1px solid var(--button-primary-focus-border);}.css-kjqv3l-ButtonLink:active{background-color:var(--button-primary-active);border:1px solid var(--button-primary-active-border);color:#fff;}",
        }
      }
      data-emotion="css kjqv3l-ButtonLink"
    />
    <a
      className="css-kjqv3l-ButtonLink"
    >
      Submit
    </a>
  </div>,
]
`;

exports[`"ButtonLink" renders in loading state 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-7uml9n-StyledAppContainer{background:#edeff0;-webkit-transition:background 0.15s;transition:background 0.15s;}",
      }
    }
    data-emotion="css 7uml9n-StyledAppContainer"
  />,
  <div
    className="css-7uml9n-StyledAppContainer"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": "html{line-height:1.15;-webkit-text-size-adjust:100%;}body{margin:0;}main{display:block;}h1{font-size:2em;margin:0.67em 0;}hr{box-sizing:content-box;height:0;overflow:visible;}pre{font-family:monospace,monospace;font-size:1em;}a{background-color:transparent;}abbr[title]{border-bottom:none;-webkit-text-decoration:underline;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted;}b,strong{font-weight:bolder;}code,kbd,samp{font-family:monospace,monospace;font-size:1em;}small{font-size:80%;}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline;}sub{bottom:-0.25em;}sup{top:-0.5em;}img{border-style:none;}button,input,optgroup,select,textarea{font-family:inherit;font-size:100%;line-height:1.15;margin:0;}button,input{overflow:visible;}button,select{text-transform:none;}button,[type="button"],[type="reset"],[type="submit"]{-webkit-appearance:button;}button::-moz-focus-inner,[type="button"]::-moz-focus-inner,[type="reset"]::-moz-focus-inner,[type="submit"]::-moz-focus-inner{border-style:none;padding:0;}button:-moz-focusring,[type="button"]:-moz-focusring,[type="reset"]:-moz-focusring,[type="submit"]:-moz-focusring{outline:1px dotted ButtonText;}fieldset{padding:0.35em 0.75em 0.625em;}legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal;}progress{vertical-align:baseline;}textarea{overflow:auto;}[type="checkbox"],[type="radio"]{box-sizing:border-box;padding:0;}[type="number"]::-webkit-inner-spin-button,[type="number"]::-webkit-outer-spin-button{height:auto;}[type="search"]{-webkit-appearance:textfield;outline-offset:-2px;}[type="search"]::-webkit-search-decoration{-webkit-appearance:none;}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit;}details{display:block;}summary{display:-webkit-box;display:-webkit-list-item;display:-ms-list-itembox;display:list-item;}template{display:none;}[hidden]{display:none;}*{box-sizing:border-box;}b,strong{font-weight:600;}body{--accent-color:#0667c8;-webkit---accent-color-highlight:#e7f0fa;--accent-color-highlight:#e7f0fa;--accent-color-highlight-inversed:#022950;--accent-color-border:#0667c8;--accent-color-focus:#3885d3;--icon-primary-active-fill:#0667c8;--icon-secondary-active-border:transparent;--button-primary-hover:#0552a0;--button-primary-active:#043e78;--button-primary-active-border:#043e78;--button-primary-focus-border:#043e78;-moz-osx-font-smoothing:grayscale;-webkit-font-smoothing:antialiased;background:#edeff0;color:rgb(29, 30, 31);display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;font-family:-apple-system,BlinkMacSystemFont,Segoe UI,Roboto,Helvetica Neue,Helvetica,Arial,sans-serif;font-weight:300;line-height:1.5;min-height:100vh;-webkit-transition:background 0.15s;transition:background 0.15s;}body,body.theme-default{--checkbox-background:#edeff0;--checkbox-background-disabled:#fafafa;--checkbox-background-disabled-selected:#9fa1a7;--checkbox-border:#54585f;--checkbox-border-disabled:#676b71;--icon-button-primary-enabled-bg:#fff;--icon-button-primary-hover-bg:#edeff0;--icon-button-primary-border:#dce0e3;--icon-button-primary-disabled-bg:#edeff0;--icon-button-primary-disabled-border:#dce0e3;--icon-button-primary-hover-border:#cbced1;--button-primary-disabled-bg:#cbced1;--button-primary-disabled-text:#54585f;--button-tertiary-bg:#fff;--button-tertiary-border:#dce0e3;--button-tertiary-hover-bg:#edeff0;--button-tertiary-hover-border:#cbced1;--button-tertiary-disabled-bg:#edeff0;--button-tertiary-disabled-border:transparent;--text-input-background:#fff;--text-input-border:#dce0e3;--text-input-border-hover:#9fa1a7;--text-input-placeholder:#676b71;--text-input-disabled:#edeff0;--text-input-label:#54585f;--select-focused-description:#fff;--danger-color:#c20013;--app-bg:#fafafa;--main-color:#000;}body.theme-dark{--checkbox-background:#34373d;--checkbox-background-disabled:#34373d;--checkbox-background-disabled-selected:#54585f;--checkbox-border:#9fa1a7;--checkbox-border-disabled:#9fa1a7;--icon-button-primary-enabled-bg:#34373d;--icon-button-primary-hover-bg:#54585f;--icon-button-primary-border:#17181a;--icon-button-primary-disabled-bg:#26272c;--icon-button-primary-disabled-border:#34373d;--icon-button-primary-hover-border:#676b71;--button-primary-disabled-bg:#676b71;--button-primary-disabled-text:#000;--button-tertiary-bg:#34373d;--button-tertiary-border:#17181a;--button-tertiary-hover-bg:#54585f;--button-tertiary-hover-border:#54585f;--button-tertiary-disabled-bg:#26272c;--button-tertiary-disabled-border:transparent;--text-input-background:#000;--text-input-border:#54585f;--text-input-border-hover:#dce0e3;--text-input-placeholder:#9fa1a7;--text-input-disabled:#17181a;--text-input-label:#dce0e3;--select-focused-description:#dce0e3;--danger-color:#ff7770;--app-bg:#26272c;--main-color:#fff;}html{background:#edeff0;-webkit-transition:background 0.15s;transition:background 0.15s;}p{margin-top:0;}",
        }
      }
      data-emotion="css-global 11ozpjr-#54585f-#dce0e3-globalStyle"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-kjqv3l-ButtonLink{color:var(--text-input-background);display:-webkit-inline-box!important;display:-webkit-inline-flex!important;display:-ms-inline-flexbox!important;display:inline-flex!important;font-size:16px;font-weight:600;overflow:hidden;text-align:center;text-overflow:ellipsis;text-transform:none;white-space:nowrap;border:0;cursor:pointer;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;margin-bottom:16px;padding:0 16px;outline:none;-webkit-text-decoration:none;text-decoration:none;touch-action:manipulation;-webkit-transition:all 0.24s;transition:all 0.24s;width:auto;border-radius:16px;height:48px;line-height:48px;max-width:100%;min-width:125px;background-color:var(--accent-color);}.css-kjqv3l-ButtonLink:hover,.css-kjqv3l-ButtonLink:focus{background-color:var(--button-primary-hover);}.css-kjqv3l-ButtonLink:focus{border:1px solid var(--button-primary-focus-border);}.css-kjqv3l-ButtonLink:active{background-color:var(--button-primary-active);border:1px solid var(--button-primary-active-border);color:#fff;}",
        }
      }
      data-emotion="css kjqv3l-ButtonLink"
    />
    <a
      className="css-kjqv3l-ButtonLink"
    >
      <svg
        fill="none"
        height={30}
        strokeWidth="3"
        style={
          {
            "display": "flex",
            "margin": "auto",
          }
        }
        viewBox="0 0 43 43"
        width={30}
      >
        <circle
          cx="21.5"
          cy="21.5"
          r="20"
          stroke="rgba(255, 255, 255, 0.07999999999999996)"
        />
        <style
          dangerouslySetInnerHTML={
            {
              "__html": ".css-1ov5q7j-Loading{-webkit-animation:animation-146xkpe 2400ms ease-in-out infinite,animation-siltbl 2400ms linear -1800ms infinite;animation:animation-146xkpe 2400ms ease-in-out infinite,animation-siltbl 2400ms linear -1800ms infinite;stroke-dasharray:125.68;transform-origin:50% 50%;}@-webkit-keyframes animation-146xkpe{0%{stroke-dashoffset:251.36;}100%{stroke-dashoffset:0;}}@keyframes animation-146xkpe{0%{stroke-dashoffset:251.36;}100%{stroke-dashoffset:0;}}@-webkit-keyframes animation-siltbl{0%{-webkit-transform:rotate(0);-moz-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}100%{-webkit-transform:rotate(360deg);-moz-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg);}}@keyframes animation-siltbl{0%{-webkit-transform:rotate(0);-moz-transform:rotate(0);-ms-transform:rotate(0);transform:rotate(0);}100%{-webkit-transform:rotate(360deg);-moz-transform:rotate(360deg);-ms-transform:rotate(360deg);transform:rotate(360deg);}}",
            }
          }
          data-emotion="css 1ov5q7j-Loading animation-146xkpe animation-siltbl"
        />
        <circle
          className="css-1ov5q7j-Loading"
          cx="21.5"
          cy="21.5"
          r="20"
          stroke="#fff"
          strokeDasharray={125.68}
          strokeLinecap="round"
        />
      </svg>
    </a>
  </div>,
]
`;
